test/models/UserModel/test.js

```diff
@@ -0,0 +1,322 @@
+      const { users } = await UserModel.search(options);
+      expect(Array.isArray(users)).to.be.true;
+    });
+    it('Should check that the error exists', async () => {
+      const { error } = await UserModel.search({
+        test: _.random(0, 50),
+        host: faker.random.string(),
+        skip: user,
+      });
+      expect(error).to.be.exist;
+    });
+  });
+  describe('On find', async () => {
+    let name;
+    let user;
+    let userAlias;
+    let condition;
+    let usersCount;
+    beforeEach(async () => {
+      await dropDatabase();
+      name = faker.name.firstName();
+      userAlias = faker.lorem.word();
+      user = await UsersFactory.Create({ name, alias: userAlias });
+      usersCount = _.random(1, 10);
+      for (let iter = 0; iter < usersCount; iter++) {
+        await UsersFactory.Create();
+      }
+      condition = { alias: userAlias };
+    });
+    it('Should return an array', async () => {
+      const { usersData } = await UserModel.find({
+        condition,
+        select: { alias: user.alias },
+        skip: 0,
+      });
+      expect(Array.isArray(usersData)).to.be.true;
+    });
+    it('Should return correct user', async () => {
+      const { usersData } = await UserModel.find({
+        condition,
+        select: { alias: user.alias },
+        skip: 0,
+      });
+      expect(usersData[0].alias).to.be.eq(userAlias);
+    });
+    it('Should return an empty array if user not found', async () => {
+      const wrongName = faker.name.firstName();
+      const { usersData } = await UserModel.find({
+        condition: { name: wrongName },
+        select: { name: wrongName },
+        sort: 'name',
+        skip: 0,
+        limit: 10,
+      });
+      expect(_.isEmpty(usersData)).to.be.true;
+    });
+    it('Should check that the error exists', async () => {
+      const { error } = await UserModel.find({
+        test: _.random(0, 50),
+        host: faker.random.string(),
+        skip: name,
+      });
+      expect(error).to.be.exist;
+    });
+  });
+  describe('On findWithSelect', async () => {
+    let usersCount;
+    beforeEach(async () => {
+      await dropDatabase();
+      usersCount = _.random(1, 10);
+      for (let iter = 0; iter < usersCount; iter++) {
+        await UsersFactory.Create({ name: `asdf_${faker.name.firstName()}` });
+      }
+    });
+    it('Should return an array', async () => {
+      const { result } = await UserModel.findWithSelect({},
+        'name, alias');
+      expect(Array.isArray(result)).to.be.true;
+    });
+    it('Should return correct user', async () => {
+      await UsersFactory.Create();
+      const { result } = await UserModel.findWithSelect({ name: { $in: [new RegExp('^asdf_')] } },
+        'name, alias');
+      expect(result.length).to.be.eq(usersCount);
+    });
+    it('Should check that the error exists', async () => {
+      const { error } = await UserModel.findWithSelect([]);
+      expect(error).to.be.exist;
+    });
+  });
```

Comment:
you do not check that after the search, only those fields that you need are selected, and this is what this method was created for